### PR TITLE
Store channel `fake_scid` with `channels`

### PR DESCRIPTION
**Pull request by da-kami**
fixes https://github.com/get10101/10101/issues/960

Since we want to record the traders *trying* top open a channel as well, we are storing he channel when we prepare the jit channel and hand the user the fake short channel id (scid).
Upon intercepting the channel we update the record if we are able to load a channel for the scid. In case we don't find a channel we print a warning (or error in case we run into an error in the db) but still proceed with opening the channel.

---

I have not implemented storing the fake short channel id in the app, because I don't think it would be very useful. 
If you think this would be good to have please let me know and I'll add it.

### CODE REVIEW

File: coordinator/migrations/2023-07-27-020530_intercept_id_channels/up.sql
Comment (holzeis) [inline coordinator/migrations/2023-07-27-020530_intercept_id_channels/up.sql:5]: 🔧 since we are querying for it, we probably should add an `INDEX` for the fake_scid.

File: crates/ln-dlc-node/src/ln/event_handler.rs
Comment (holzeis) [inline crates/ln-dlc-node/src/ln/event_handler.rs:703]: nit: useless newline 🙃

File: crates/ln-dlc-node/src/node/invoice.rs
Comment (holzeis) [inline crates/ln-dlc-node/src/node/invoice.rs:178]: 🤔 That would create a channel in state `Pending`. Shouldn't we introduce a dedicated state for channels that are announced through a created invoice? I feel like we will get a lot of those records 🙈.

e.g. `Announced`

File: crates/ln-dlc-node/src/node/invoice.rs
Comment (holzeis) [inline crates/ln-dlc-node/src/node/invoice.rs:155]: 🔧 We are still creating interceptable invoices for every invoice generated by the app, regardless if a channel exists or not. That may lead to a lot of channel entries that would incorrectly indicate that funding the channel failed.

We should probably add here a check if the user already has a channel. This will still not be 100% correct, but I think will work for the most cases.

Comment (da-kami) [reply crates/ln-dlc-node/src/node/invoice.rs:155]: Do I understand you correctly, that we use this code path for every invoice and not just for opening a channel? 
If so, then I would probably move this code so that we can just do it upon channel creation attempt and not for all invoices.

File: crates/ln-dlc-node/src/node/invoice.rs
Comment (da-kami) [reply crates/ln-dlc-node/src/node/invoice.rs:178]: That's reasonable I think.

File: crates/ln-dlc-node/src/node/invoice.rs
Comment (da-kami) [reply crates/ln-dlc-node/src/node/invoice.rs:155]: Can confirm that all invoices are created by calling the `prepare_jit_channel` endpoint. 

I thought about changing this to two separate endpoints, one for initial channel creation and one for consecutive invoices. But I came to the conclusion that it's better to role with your proposal of just adding a check in the coordinator that decides if this is an initial channel open or not. That gives the coordinator more control which is good.

Might still be a good idea to rename `prepare_jit_channel` to a more intuitive name for what it is used for.

File: mobile/native/src/ln_dlc/node.rs
Comment (luckysori) [inline mobile/native/src/ln_dlc/node.rs:418]: 👍I think this is okay, although it reminds me of how much I hate the `Storage` trait I added.

File: coordinator/src/db/channels.rs
Comment (luckysori) [inline coordinator/src/db/channels.rs:145]: 🙃 Super nit.
```suggestion
            ln_dlc_node::channel::ChannelState::Announced => ChannelState::Announced,
            ln_dlc_node::channel::ChannelState::Pending => ChannelState::Pending,
```

File: crates/ln-dlc-node/src/channel.rs
Comment (luckysori) [inline crates/ln-dlc-node/src/channel.rs:121]: 🔧 We could comment that this is only `Some` if we are dealing with a JIT channel.

File: crates/ln-dlc-node/src/node/invoice.rs
Comment (luckysori) [inline crates/ln-dlc-node/src/node/invoice.rs:165]: 🔧 I would just fail here. Obviously storing the fake SCID is not super crucial, but I don't see why it should fail anyway.

File: mobile/native/src/db/models.rs
Comment (luckysori) [inline mobile/native/src/db/models.rs:1144]: 🤔 I think we do have this value in `value`.

Comment (da-kami) [reply mobile/native/src/db/models.rs:1144]: This is on the app, and on the app we don't have it; the model was not extended to reflect this value.

File: crates/ln-dlc-node/src/ln/event_handler.rs
Comment (da-kami) [reply crates/ln-dlc-node/src/ln/event_handler.rs:711]: Early in the process, like the scenario below, I think it's OK to fail. But here it would be a shame if we have to fail creation the channel because we don't find the `scid` or there is an error. I would keep this; I don't think it does any harm.

File: crates/ln-dlc-node/src/node/invoice.rs
Comment (da-kami) [reply crates/ln-dlc-node/src/node/invoice.rs:165]: I wonder if the above check is exhaustive enough to say we fail here. I wonder if we can have cases where we don't have a usable channel *just yet*; then the invoice creation would fail (because we cannot insert the same channel twice) event though the user just wants to make a payment. I felt it might be better to not make the invoices creation depend on this insert...

Comment (da-kami) [reply crates/ln-dlc-node/src/node/invoice.rs:165]: I think eventually it would be good to have two different pathes for initial channel open and payments over an existing channel. But I did not want to refactor this as part of this PR...

File: mobile/native/src/db/models.rs
Comment (luckysori) [reply mobile/native/src/db/models.rs:1144]: Ah, good point. Not ideal to have a shared model, but we can clean that up at some point in the future.

File: crates/ln-dlc-node/src/node/invoice.rs
Comment (luckysori) [reply crates/ln-dlc-node/src/node/invoice.rs:165]: I see where you are coming from.

I think ideally we would only call this function when we know that the invoice will be used to open a JIT channel. Then we wouldn't need to reason about the possibility that this might be for a regular payment.

It's probably okay to leave this as is for now then.

**Review by luckysori — APPROVED** [review]
LGTM.

My only disagreement is on just logging errors in the event handler. I think it's a bit better to error because the feature we're adding should just work (even if it's not super crucial). In theory we could miss problems with the feature if the tests pass with these errors being logged.

**Comment by luckysori** [comment]
Assigning myself because we still have a merge freeze and @da-kami is leaving us 😢

**Comment by luckysori** [comment]
Let's wait for the release of version 1.2 from `main` before merging this.

**Pull request merged**
